paper-calendar.html

```diff
@@ -652,7 +669,27 @@
       },
       _withinValidRange: function(date) {
         if (this._isValidDate(date)) {
-          return (!this.minDate || date >= this.minDate) && (!this.maxDate || date <= this.maxDate);
+          return  (!this.minDate || date >= this.minDate) && (!this.maxDate || date <= this.maxDate);
+        }
+        return false;
+      },
+
+      _confirmsToWhiteAndBlacklist: function(date){
```

Comment:
Collapse these 3 into:
```
/**
 * Ensures that our date is valid and passes both white and black list checks.
 * @protected
 *
 * @param {Date} date
 * @return {Boolean}
 */
_checkWhiteBlackLists: function(date) {
  return this._isValidDate(date) &&
    (this.validDates || []).indexOf(this.dateFormat(date,'YYYY-MM-DD')) < 0 &&
    (this.invalidDates || []).indexOf(this.dateFormat(date,'YYYY-MM-DD')) < 0;
}
```
So we aren't biting the cost of always checking both lists, and checking for a valid date 3 times when we don't have to.